Main/ReportWindow.cpp

```diff
@@ -0,0 +1,106 @@
+/*
+ */
+
+#include <Alert.h>
+#include <Application.h>
+#include <Box.h>
+#include <Button.h>
+#include <Catalog.h>
+#include <ColumnListView.h>
+#include <ColumnTypes.h>
+#include <Font.h>
+#include <LayoutBuilder.h>
+#include <Menu.h>
+#include <MenuBar.h>
+#include <MenuItem.h>
+#include <Roster.h>
+#include <StringView.h>
+#include <TextControl.h>
+#include <View.h>
+
+#include <string.h>
+#include <stdlib.h>
+
+#include "constants.h"
+#include "functions.h"
+#include "PreviewColumn.h"
+#include "ReportWindow.h"
+
+#undef B_TRANSLATION_CONTEXT
+#define B_TRANSLATION_CONTEXT "ReportWindow"
+
+
+ReportWindow::ReportWindow(BList *filelist) : BWindow( BRect( 60, 80, 600, 280), B_TRANSLATE("Report"),
+	 B_TITLED_WINDOW,B_AUTO_UPDATE_SIZE_LIMITS)
+{
+
+	BStringView *messageView = new BStringView("Errors", B_TRANSLATE("There were some problems during the renaming"));
+	BBox* reportBox = new BBox("Errors");
+	reportBox->SetLabel(B_TRANSLATE("There were some problems during the renaming"));
+	messageView->SetExplicitMaxSize(BSize(B_SIZE_UNLIMITED, B_SIZE_UNSET));
+	BColumnListView* reportView = new BColumnListView("reportView", B_FRAME_EVENTS|B_NAVIGABLE, B_NO_BORDER);
+
+	//info columns
+	int32 i = 0;
+	reportView->AddColumn(new BStringColumn(B_TRANSLATE("Name"), 165, 10, 600, B_TRUNCATE_END), i++);
+	reportView->AddColumn(new PreviewColumn(B_TRANSLATE("Failed Rename"), 165, 10, 600, B_TRUNCATE_END), i++);
+	reportView->AddColumn(new BStringColumn(B_TRANSLATE("Problem"), 185, 10, 600, B_TRUNCATE_END), i++);
+	
+	FileListItem	*listItem;
+	for (int32 i = 0; (listItem = (FileListItem *)filelist->ItemAt(i)) != NULL; i++ ) {
+		if (listItem->Error() != 0 && listItem->fNewName != "")
+			{
+				BRow* row = new BRow();
+				PreviewField* previewField = new PreviewField(listItem->fNewName);
+				previewField->SetError(true);
+				int32 i = 0;
+				row->SetField(new BStringField(listItem->fName), i++);
+				row->SetField(previewField, i++);
+				BString errorMsg;
+				switch (listItem->Error()) {
+					case 0 : // this should never happen
+						break; 
+					case 1 : errorMsg = B_TRANSLATE("Duplicated name");
```

Comment:
Why is there a different message for case 1 and case 2?

Reply:
we spoke about this alot of time
1) is the duplicated already discovered by the application
2) this error happen when the application finds a duplicate on disk

Reply:
OK. But why is there a distinction if the reason is for both "File with that name already exists"?